### PR TITLE
Validation rules

### PR DESCRIPTION
**Pull request by motdotla**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -3,18 +3,25 @@ dotenv.load();
...
+var SendGrid 	= sg.SendGrid;
+Email			= sg.Email;
+sendgrid    	= new SendGrid(sendgrid_username, sendgrid_password);
+
+fs				= require('fs');
 
 var port        = parseInt(process.env.PORT) || 3000;
 var Hapi        = require('hapi');
-server          = new Hapi.Server(+port, '0.0.0.0', { cors: true });
+server     		= new Hapi.Server(+port, '0.0.0.0', { cors: { origin: [domain] } });
```
Comment (motdotla) [inline app.js]: good call

File: routes/send.js [hunk 2]
```diff
@@ -1,30 +1,67 @@
...
+
+  // Load All Rules
+  if(fs.existsSync("./rules/all.js")){
+    var rules = require("./../rules/all");
+  }else{
+    var rules = {};
+  }
+
+  // Load Endpoint Rules
+  if(request.params.endpoint && fs.existsSync("./rules/" + request.params.endpoint + ".js")){
```
Comment (motdotla) [inline routes/send.js:17]: cool

**Comment by motdotla** [comment]
Looks good @nquinlan. I think we're putting something together here that will be really useful for people. Keep it up!

The only recommendation I have is to update to the latest 0.3.0-rc.1.0 sendgrid nodejs module. Then we get a chance to test that in live action before it becomes official 0.3.0.

**Comment by nquinlan** [comment]
I think you're right, we should most definitely update to `0.3.0-rc.1.0`!

I'm also in the process of creating a GUI editor for the config files so it'll be _really_ easy for someone to go ahead and create an email service.

Another thing I think we should put on the roadmap is per IP usage limits (e.g. no more than 10 emails per min from each IP) to prevent abuse.

**Comment by motdotla** [comment]
Old. Closing.

**Pull request closed without merging**